src/pages/home/report/ReportActionsList.js

```diff
@@ -288,9 +288,13 @@ function ReportActionsList({
             if (!currentUnreadMarker) {
                 const nextMessage = sortedReportActions[index + 1];
                 const isCurrentMessageUnread = isMessageUnread(reportAction, report.lastReadTime);
-                shouldDisplay = isCurrentMessageUnread && !isMessageUnread(nextMessage, report.lastReadTime);
-                if (!messageManuallyMarkedUnread) {
-                    shouldDisplay = shouldDisplay && reportAction.actorAccountID !== Report.getCurrentUserAccountID();
+                const isNextMessageRead = !isMessageUnread(nextMessage, report.lastReadTime);
+                const isWithinVisibleThreshold = scrollingVerticalOffset.current < MSG_VISIBLE_THRESHOLD ? reportAction.created < userActiveSince.current : true;
+
+                shouldDisplay = isCurrentMessageUnread && isNextMessageRead && isWithinVisibleThreshold;
+
+                if (shouldDisplay && !messageManuallyMarkedUnread) {
+                    shouldDisplay = reportAction.actorAccountID !== Report.getCurrentUserAccountID();
```

Comment:
Should we ignore this condition if shoudlDisplay was set as false before? What is the purpose?

Reply:
@barttom  Yes, the previous logic worked the same (if shoudlDisplay was false before, it was also false for this condition as well), the code was like that:

https://github.com/Expensify/App/blob/39d2a8457897439ebab57cf446fc23e90c553a8d/src/pages/home/report/ReportActionsList.js#L292-L294